go/gossip/handler.go

```diff
@@ -66,13 +85,29 @@ func (h *Handler) HandleSTHPollination(rw http.ResponseWriter, req *http.Request
 		return
 	}
 
+	sthToKeep := make([]ct.SignedTreeHead, 0, len(p.STHs))
+	for _, sth := range p.STHs {
+		v, found := h.verifiers[sth.LogID]
+		if !found {
+			log.Printf("Pollination entry for unknown logID: %s", sth.LogID.Base64String())
+			continue
+		}
+		if err := v.VerifySTHSignature(sth); err != nil {
+			log.Printf("Failed to verify STH, dropping: %v", err)
+			continue
+		}
+		sthToKeep = append(sthToKeep, sth)
+	}
+	p.STHs = sthToKeep
```

Comment:
Ah, I missed the source, I thought it was coming from somewhere else, for some reason. Erasing while iterating is always a bit iffy, fine like this.